x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/action/GetDataFrameTransformsStatsAction.java

```diff
@@ -174,15 +193,15 @@ public void readFrom(StreamInput in) {
         public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
             builder.startObject();
             toXContentCommon(builder, params);
-            builder.field(DataFrameField.COUNT.getPreferredName(), transformsStateAndStats.size());
+            builder.field(DataFrameField.COUNT.getPreferredName(), totalCount == 0 ? transformsStateAndStats.size() : totalCount);
```

Comment:
Can "0" be a valid value of totalCount? If so, totalCount should likely be nullable.

Reply:
It will always be the case that `totalCount <= transformsStateAndStats.size()`. 

I see what your concern, but given how this class is used, there are no use cases where `totalCount` is `< transformsStateAndStats.size()`. I will try to make this more obvious and safe in the class.